### ISSUE TITLE
Wrong multiplier value in retry delay function example

### ISSUE THREAD

**Issue by zerda** [confluentinc/parallel-consumer#622]
Hi, there is a typo in the README.md file at [11.1. Retry Delay Function](https://github.com/confluentinc/parallel-consumer#retry-delay-function).

```java
final double multiplier = 0.5;
final int baseDelaySecond = 1;

ParallelConsumerOptions.<String, String>builder()
        .retryDelayProvider(recordContext -> {
            int numberOfFailedAttempts = recordContext.getNumberOfFailedAttempts();
            long delayMillis = (long) (baseDelaySecond * Math.pow(multiplier, numberOfFailedAttempts) * 1000);
            return Duration.ofMillis(delayMillis);
        });
```

While the `multiplier` is set to 0.5, it results in delay being always less than 1s, and decreasing at every call. This does not adhere to the exponential backoff strategy.
| Failed Attempts 	| Delay (ms)         	|
|-----------------	|--------------------	|
| 1               	| `1*0.5^1*1000 = 500` 	|
| 2               	| `1*0.5^2*1000 = 250` 	|
| 3               	| `1*0.5^3*1000 = 125` 	|

**Comment by lutzh** [comment]
It seems this has been fixed in https://github.com/confluentinc/parallel-consumer/pull/864, so maybe this issue should be closed?

**Comment by singhvishalkr** [comment]
This appears to already be resolved: the retry-delay example in the rendered README (which comes from [`README.adoc`](https://github.com/confluentinc/parallel-consumer/blob/master/README.adoc#L797)) now uses `multiplier = 2.0`, which gives the correct exponential backoff (500 ms → 2 s → 4 s → 8 s, etc.):

```java
final double multiplier = 2.0;
final int baseDelaySecond = 1;

ParallelConsumerOptions.<String, String>builder()
        .retryDelayProvider(recordContext -> {
            int numberOfFailedAttempts = recordContext.getNumberOfFailedAttempts();
            long delayMillis = (long) (baseDelaySecond * Math.pow(multiplier, numberOfFailedAttempts) * 1000);
            return Duration.ofMillis(delayMillis);
        });
```

The change landed in #864 (_"docs: fix wrong multiplier value in retry delay function example"_, merged 2025-10-27).

Since the reported README source no longer contains the `0.5` multiplier, this issue can probably be closed as **completed**. Happy to confirm with a maintainer if the author of #864 didn't originally link it to this issue.

cc @zerda (original reporter) — does the current README section look correct to you now?

**Comment by astubbs** [comment]
Tracked in the maintained fork as astubbs/parallel-consumer#167, which carries a summary of this
issue and its current status there.

The **Confluent-hosted version** of this project is no longer maintained - its README says so.
**Parallel Consumer itself continues** in my fork: https://github.com/astubbs/parallel-consumer

Nothing in this repository is closed or changed by this comment.

**Comment by astubbs** [comment]
Worth a check before this is closed as completed: #864 patched `README.adoc`, which is a generated
file. The snippet's source is `CoreApp.java`, included into the docs via `tag::customRetryDelay[]`,
and it still reads `final double multiplier = 0.5` on master today.

So the rendered README shows `2.0` while the code it is generated from shows `0.5`, and the next
`mvn process-sources` reverts it.

Not hypothetical: the maintained fork inherited that `2.0`, then a routine README regeneration on
2026-04-09 restored `0.5` - twelve days before the comment above.

Full analysis and a source-level fix: astubbs/parallel-consumer#167 and astubbs/parallel-consumer#196.

(Minor: with `2.0` the sequence is 1s, 2s, 4s, 8s - `getNumberOfFailedAttempts()` starts at 0, so the
first term is `2^0`, not 500ms.)

**Comment by astubbs** [comment]
@zerda, @lutzh, @singhvishalkr - fixed and released in the maintained fork, in [v0.6.0.0](https://github.com/astubbs/parallel-consumer/releases/tag/v0.6.0.0). The coordinates changed with the fork - on Maven Central it is now:

```xml
<dependency>
    <groupId>bz.stub.parallelconsumer</groupId>
    <artifactId>parallel-consumer-core</artifactId>
    <version>0.6.0.0</version>
</dependency>
```

(`parallel-consumer-vertx`, `parallel-consumer-reactor` and `parallel-consumer-mutiny` likewise.) The Java package is `bz.stub.parallelconsumer` too, so imports change with it - one command rewrites them (GNU `sed`; on macOS use `sed -i ''`):

```bash
find . -name '*.java' -exec sed -i 's/io\.confluent\.parallelconsumer/bz.stub.parallelconsumer/g' {} +
```

The short version of the fix: the README's custom retry-delay example halved the wait on every failed attempt instead of growing it; the multiplier is corrected in the template the README is generated from, so the docs-only patch offered here cannot be lost again (astubbs/parallel-consumer#196). The fork's mirror is astubbs/parallel-consumer#167.
